### PR TITLE
Remove obselete memoryview getbuffer fallback code

### PR DESCRIPTION
**Pull request by da-woods**
It appears to have been be used for Python 2.6 (which I don't
think has the buffer protocol). Most of the mechanism had already
been removed, but the capsule attributes that it relied on were
still generated.

Also removed some utility code that was only used for this
feature

### CODE REVIEW

**Comment by da-woods** [comment]
The issues look to be warnings about `DEF` in the memoryview code that now appear on the wrong line. I'll fix those later

**Comment by scoder** [comment]
> Python 2.6 (which I don't think has the buffer protocol)

It did. Python 2.6 was the version that went with Python 3.0/3.1, and the buffer protocol was already in there.
But we no longer need to care about 2.6 for Cython 3.0 (and 0.29.x only 'officially' still supports it – its usage is negligible).

File: Cython/Compiler/MemoryView.py [hunk 1]
```diff
@@ -851,8 +851,7 @@ def use_cython_array_utility_code(env):
                   memviewslice_init_code,
                   is_contig_utility,
                   overlapping_utility,
-                  copy_contents_new_utility,
-                  ModuleNode.capsule_utility_code],
+                  copy_contents_new_utility],
```
Comment (scoder) [inline Cython/Compiler/MemoryView.py:854]:
```suggestion
                  copy_contents_new_utility,
                  ],
```

**Comment by scoder** [comment]
According to code comments, this was meant to support pre-2.6 (back in 2012), which really didn't have the new buffer protocol. So yes, this can go. Thanks for removing it.

**Pull request merged**